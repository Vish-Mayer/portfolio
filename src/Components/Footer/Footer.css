.main-footer{
  background: rgb(35, 34, 39);
  color: #fff;
  padding-top: 1em;
  padding-bottom: 1em;
}

.main-footer{
  display: flex;
  flex-wrap: wrap;
}

.media-img{
  max-width: 100%;
  max-height: 100%;
}


.media-btn{
  height: 100px;
  width: 100px;
  overflow: hidden;
  background: rgb(4, 80, 105);
  border: none;
  border-radius: 25px;
}

.footer-end{
  left: 50%;
  padding-top: 2em;
  margin: 0 auto;
}

.created-date{
  color: rgb(221, 62, 62);
}

.media-btn:hover > .media-img{
  animation: shake 0.82s cubic-bezier(.36,.07,.19,.97) both;
  transform: translate3d(0, 0, 0);
  backface-visibility: hidden;
  perspective: 1000px;
}

@keyframes shake {
  10%, 90% {
    transform: translate3d(-1px, 0, 0);
  }
  
  20%, 80% {
    transform: translate3d(2px, 0, 0);
  }

  30%, 50%, 70% {
    transform: translate3d(-4px, 0, 0);
  }

  40%, 60% {
    transform: translate3d(4px, 0, 0);
  }
}
