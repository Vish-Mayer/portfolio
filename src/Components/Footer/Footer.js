import React from 'react';
import './Footer.css'

import media1 from '../../Assets/Icons/linkedinS.png'
import media2 from '../../Assets/Icons/github.png'
import media3 from '../../Assets/Icons/cvlogo2.png'
import Doc from '../../Assets/Documents/Vishal Kakkad CV 2020.pdf'

function Footer() {
  return(
    <div className="main-footer">
      <div className="container-fluid p-0 d-flex justify-content-center">
        <div className="media-row row p-0">

          <div className="col-md-4 col-4">
            <a href="https://www.linkedin.com/in/vishal-mayer-kakkad-a04607189/" target="_blank" rel="noreferrer noopener">
              <button 
              className="media-btn" 
              title="Linkedin">
                <img className="media-img" src={media1} alt="media1"/>
              </button>
            </a>
          </div>

          <div className="col-md-4 col-4">
          <a href="https://github.com/Vish-Mayer/" target="_blank" rel="noreferrer noopener">
            <button
            className="media-btn" 
            title="Github">
              <img className="media-img" src={media2} alt="media2"/>
            </button>
          </a> 
          </div>

          <div className="col-md-4 col-4">
            <a href={Doc} target="_blank" rel="noreferrer noopener">
              <button 
              className="media-btn" 
              title="Download CV">
                <img className="media-img" src={media3} alt="media3"/>
              </button>
            </a>
          </div>

        </div>
      </div>
      <div className='footer-end d-flex'>
      <div className="created-by">Vishal Mayer Kakkad </div>
      <div className="created-date">&nbsp;@2020</div>
      </div>
    </div>
  )
}
export default Footer