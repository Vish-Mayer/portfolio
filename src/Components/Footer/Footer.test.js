import React from 'react'
import Footer from './Footer';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Footer', () => {

  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Footer/>)
  });

  it('it has a row of media buttons', () => {
    expect(wrapper.find('.media-row').props()).toMatchSnapshot()
  });
});