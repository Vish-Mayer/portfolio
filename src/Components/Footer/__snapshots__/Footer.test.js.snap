// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer it has a row of media buttons 1`] = `
Object {
  "children": Array [
    <div
      className="col-md-4 col-4"
    >
      <button
        className="media-btn"
      >
        <img
          alt="media1"
          className="media-img"
          src="linkedinS.png"
        />
      </button>
    </div>,
    <div
      className="col-md-4 col-4"
    >
      <button
        className="media-btn"
      >
        <img
          alt="media2"
          className="media-img"
          src="github.png"
        />
      </button>
    </div>,
    <div
      className="col-md-4 col-4"
    >
      <button
        className="media-btn"
      >
        <img
          alt="media3"
          className="media-img"
          src="cvlogo2.png"
        />
      </button>
    </div>,
  ],
  "className": "media-row row p-0",
}
`;
